### DISCUSSION TITLE
Oauth2 New User Management

### DISCUSSION THREAD

**Discussion by applesoff** [gotson/komga#1563] [Ideas]
### Describe your suggested feature

I have set up Google and GitHub OAuth2 for new users to make an account with the env variable KOMGA_OAUTH2_ACCOUNT_CREATION=true so that any account made does not need me to set it up in advance creating a more seamless process for friends/family when visiting the server. 

The issue is that new users made this way have access to all content. For various reasons, this is not ideal. (i.e. restricted content, 18+)

It would be great to add options in the UI so that any new account added is subjected to a standard rule set such as blocking access to certain libraries or having share tags set for new users (that can be altered after account creation as is the current way) so libraries with sensitive content cannot be seen initially unless later approved. 

### Other details

An alternative to this proposed change would be a pend status is applied to a new account added with OAuth2 and a notification (email, txt, discord push) is sent to any admin accounts for approval/denial with the ability to alter their settings prior to their access.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by applesoff** [comment]
Recent support messages reminded me of this.
If there was a default access criteria for new users that would work. For example, a setting denoting new user inclusion/exclusion sharing tags. So I can set an entire library to "No" and new users will have all series tagged "No" not show up.

**Comment by DuvelCorp** [comment]
I am having exactly the same issue with adult contain, and this prevent me to use Oauth2 user creation in Komga.
The only approach I found is to create an additional komga install not connected with Oauth2, and to move all the adult stuff into it.... but this a bit cumbersome.
So I hope you will make this enhancement happen !

**Comment by bilyboy785** [comment]
Same issue here. Any news on this feature ? Ability to set default permissions for new users ? Or ability to allow only certains libraries ?

**Comment by gerrewsb** [comment]
it would be nice if we could map groups and/or claims to Komga-permissions. but even an ability to set default permissions for new users, like suggested by other people, would be a good start.

**Comment by gotson** [comment]
In the context of my work i recently came accross https://en.wikipedia.org/wiki/System_for_Cross-domain_Identity_Management which seems to be the standard for managing account creation from a central authority. I am thinking whether it would be a better way to manage OAuth2 users, instead of the current on-the-fly creation, which lacks configuration for defaults.

Would anyone here using OAuth2 login would know whether the IdP you are using supports SCIM ? This would probably apply mostly to self-hosted IdP though.

**Reply by imi-kitten** [reply]
looks like authentik does:
https://docs.goauthentik.io/docs/add-secure-apps/providers/scim/

**Reply by bootsievlot** [reply]
Have a look at audiobookshelf. They have oidc implemented really well. You can set permissions based on tag,roles and groups(from the oidc provider) and the accounts are still created on the fly. 

The config for that looks like this. 
```
{
  "canDownload": false,
  "canUpload": false,
  "canDelete": false,
  "canUpdate": false,
  "canAccessExplicitContent": false,
  "canAccessAllLibraries": false,
  "canAccessAllTags": false,
  "canCreateEReader": false,
  "tagsAreDenylist": false,
  "allowedLibraries": [
    "5406ba8a-16e1-451d-96d7-4931b0a0d966",
    "918fd848-7c1d-4a02-818a-847435a879ca"
  ],
  "allowedTags": [
    "ExampleTag",
    "AnotherTag",
    "ThirdTag"
  ]
}
```

**Reply by gotson** [reply]
> Have a look at audiobookshelf. They have oidc implemented really well. You can set permissions based on tag,roles and groups(from the oidc provider) and the accounts are still created on the fly.

could you point to their documentation maybe ?

**Comment by sgayda2** [comment]
I have seen other services use the groups provided from oauth to determine what libraries can be accessed. So for example if my authentik instance has two groups 'family' and 'friends' and i pass those along to komga. I could in theory, use the groups to determine that anyone in the family group can see my family library as well as my extended library, and anyone in the friends group can see only the extended library.

I do like audiobookshelf, but have not tried to use their claims/groups/tags.
audiobookshelf github is: https://github.com/advplyr/audiobookshelf
audiobookshelf docs: https://www.audiobookshelf.org/guides/oidc_authentication
